src/uu/uuidgen/src/uuidgen.rs

```diff
@@ -0,0 +1,245 @@
+// This file is part of the uutils util-linux package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+#[cfg(target_os = "windows")]
+use windows::Win32::{
+    Foundation::{ERROR_BUFFER_OVERFLOW, ERROR_SUCCESS},
+    NetworkManagement::IpHelper::{
+        GetAdaptersAddresses, GAA_FLAG_SKIP_ANYCAST, GAA_FLAG_SKIP_DNS_SERVER,
+        GAA_FLAG_SKIP_FRIENDLY_NAME, GAA_FLAG_SKIP_MULTICAST, GAA_FLAG_SKIP_UNICAST,
+        IP_ADAPTER_ADDRESSES_LH,
+    },
+    Networking::WinSock::AF_UNSPEC,
+};
+
+use clap::{crate_version, Arg, ArgAction, Command};
+#[cfg(any(
```

Comment:
same
not windows!

Reply:
I was thinking this, but it seems Redox OS does not support this API. After sleeping on it, I think the best thing to do would be `#cfg(all(target_family = "unix", not(target_os = "redox")))`

Reply:
yes, please update this :)